### PR TITLE
Document using xpack.ml.use_auto_machine_memory_percent setting in docker getting started

### PR DESCRIPTION
**Pull request by davidkyle**
If `xpack.ml.use_auto_machine_memory_percent` is not explicitly set to true then the default value (false) means ML will only use 30% of the available memory making it impractical to run the ELSER model without giving the container a huge amount of memory. [Start local](https://github.com/elastic/start-local/blob/main/start-local.sh#L406) sets `xpack.ml.use_auto_machine_memory_percent : true` for this reason 

For users wanting to get started with semantic search it is confusing that they cannot run an ml model in the container as instructed in the _Install Elasticsearch with Docker_ guide. 

The single node docker instructions have been updated with a command that gives the container enough memory to run the ELSER model and enables `xpack.ml.use_auto_machine_memory_percent`.

For the multi-node guide the docker compose file is updated to enable the ml setting for every node in the cluster. An alternative would be to make only 1 of the 3 nodes an ml node and apply the setting to that node only.

### CODE REVIEW

File: docs/reference/setup/install/docker.asciidoc [hunk 1]
```diff
@@ -86,6 +86,14 @@ docker run --name es01 --net elastic -p 9200:9200 -it -m 1GB {docker-image}
 TIP: Use the `-m` flag to set a memory limit for the container. This removes the
 need to <<docker-set-heap-size,manually set the JVM size>>.
 +
+
+Machine Learning features such as semantic search with the ELSER model
+require a larger container with more than 1GB of memory.
+If you intend to use the ML capabilities start the container with this command
```
Comment (shainaraskas) [inline docs/reference/setup/install/docker.asciidoc:92]:
```suggestion
If you intend to use the machine learning capabilities, then start the container with this command:
```

File: docs/reference/setup/install/docker.asciidoc [hunk 2]
```diff
@@ -86,6 +86,14 @@ docker run --name es01 --net elastic -p 9200:9200 -it -m 1GB {docker-image}
 TIP: Use the `-m` flag to set a memory limit for the container. This removes the
 need to <<docker-set-heap-size,manually set the JVM size>>.
 +
+
+Machine Learning features such as semantic search with the ELSER model
```
Comment (shainaraskas) [inline docs/reference/setup/install/docker.asciidoc:90]:
```suggestion
Machine learning features such as <<semantic-search-elser, semantic search with ELSER>>
```

**Review by shainaraskas — APPROVED** [review]
approved with a couple of edits!

**Comment by davidkyle** [comment]
In the preview, the formatting for the code example I added is not right. I can't see what I did wrong, @shainaraskas can see what I'm missing please

Screenshot 2024-10-07 at 20 01 56

File: docs/reference/setup/install/docker.asciidoc [hunk 3]
```diff
@@ -86,6 +86,14 @@ docker run --name es01 --net elastic -p 9200:9200 -it -m 1GB {docker-image}
 TIP: Use the `-m` flag to set a memory limit for the container. This removes the
 need to <<docker-set-heap-size,manually set the JVM size>>.
 +
+
+Machine learning features such as <<semantic-search-elser, semantic search with ELSER>>
+require a larger container with more than 1GB of memory.
+If you intend to use the machine learning capabilities, then start the container with this command:
+[source,sh,subs="attributes"]
```
Comment (szabosteve) [inline docs/reference/setup/install/docker.asciidoc:93]:
```suggestion
+
[source,sh,subs="attributes"]
```

File: docs/reference/setup/install/docker.asciidoc [hunk 4]
```diff
@@ -86,6 +86,14 @@ docker run --name es01 --net elastic -p 9200:9200 -it -m 1GB {docker-image}
...
+----
+docker run --name es01 --net elastic -p 9200:9200 -it -m 6GB -e "xpack.ml.use_auto_machine_memory_percent=true" {docker-image}
+----
```
Comment (szabosteve) [inline docs/reference/setup/install/docker.asciidoc:96]:
```suggestion
----
+
```

File: docs/reference/setup/install/docker.asciidoc [hunk 5]
```diff
@@ -86,6 +86,14 @@ docker run --name es01 --net elastic -p 9200:9200 -it -m 1GB {docker-image}
 TIP: Use the `-m` flag to set a memory limit for the container. This removes the
 need to <<docker-set-heap-size,manually set the JVM size>>.
 +
+
+Machine learning features such as <<semantic-search-elser, semantic search with ELSER>>
```
Comment (szabosteve) [inline docs/reference/setup/install/docker.asciidoc:90]:
```suggestion
{ml-cap} features such as <<semantic-search-elser, semantic search with ELSER>>
```

**Review by szabosteve — APPROVED** [review]
LGTM2!

**Comment by davidkyle** [comment]
@elasticmachine update branch

**Pull request merged**